cvpy/base/ImageTable.py

```diff
@@ -128,4 +200,23 @@ def type(self) -> str:
 
     @type.setter
     def type(self, type) -> None:
+        if type not in self.table.columns:
+            raise Exception(f'Column "{type}" is not present in the table.')
         self._type = type
+
+    def as_dict(self) -> dict:
+        '''
+        Creates a dictionary representation of this object.
+        :return: A dictionary with all of the properties as keys and the property values as values
+        '''
+        d = {}
+        for k, v in vars(self).items():
+            d[k[1:]] = v
+        return d
+
+    def has_decoded_images(self) -> bool:
```

Comment:
This is really useful for Kevin's functions. Thanks for adding this!

Reply:
Ack.